test/edge.js

```diff
@@ -1,5 +1,6 @@
 const util = require('util')
```

Comment:
This entire thing is just one test with a lot of assertions. Is this how we want to test data component classes?

Reply:
lol it is not. i deemed it out of scope, however, to completely refactor the existing tests so i chose to add my tests in the same pattern as what existed and plan on a future refactor